DS13/code/modules/overmap/combat.dm

```diff
@@ -0,0 +1,507 @@
+	var/damage = 0 //How much damage do we do? This is +10'd assuming weapons are powered!
+	var/charging = FALSE //charging weapons? if so, we can't fire
+	var/admin_override = FALSE //Need to test? set this on and you can fire and move yourself
+	var/fire_mode = "phaser"
+	var/photons = 4 //How many torpedoes do we have?
+	var/list/weapon_sounds = list('DS13/sound/effects/weapons/phaser.ogg','DS13/sound/effects/weapons/phaser2.ogg','DS13/sound/effects/weapons/phaser3.ogg','DS13/sound/effects/weapons/phaser4.ogg')
+	var/obj/structure/overmap/tractor_target = null //Are we tractoring a target? This forces it to move towards us.
+	var/datum/beam/tractor_beam = null
+	var/hail_ready = TRUE //Hailing cooldown
+	var/torpedo_damage = 30
+	var/area/linked_area = null
+	var/destroyed = FALSE
+	var/damage_sector = "shields" //Tactical can change this! What system are we targeting?
+
+/obj/structure/overmap/proc/send_sound_crew(var/sound/S)
+	for(var/mob/M in operators)
+		SEND_SOUND(M, S)
+
+/obj/structure/overmap/proc/send_text_crew(var/S)
+	for(var/mob/M in operators)
+		to_chat(M, S)
+
+/obj/effect/temp_visual/ship_explosion
+	icon = 'DS13/icons/overmap/effects.dmi'
+	icon_state = "ship_explode"
+	duration = 30
+
+/obj/effect/temp_visual/ship_explosion/Initialize()
+	. = ..()
+	pixel_x = rand(0,10)
+	pixel_y = rand(0,10)
+
+/obj/structure/overmap/proc/onMouseDown(object, location, params, mob/mob)
+	var/sound/thesound = pick('DS13/sound/effects/computer/beep.ogg','DS13/sound/effects/computer/beep2.ogg','DS13/sound/effects/computer/beep3.ogg')
+	SEND_SOUND(mob, thesound)
+	if(object == src)
+		return attack_hand(mob)
+	if(istype(object, /obj/screen) && !istype(object, /obj/screen/click_catcher))
+		return
+	if(!admin_override) //Allows me to fly and shoot for testing
+		if(mob != tactical)
+			if(science)
+				if(mob == science && istype(object, /obj/structure/overmap))
+					var/list/options = list("tractor", "hail","tractor-cancel")
+					for(var/option in options)
+						options[option] = image(icon = 'DS13/icons/actions/weaponselect.dmi', icon_state = "[option]")
+					var/dowhat = show_radial_menu(mob,object,options)
+					if(!dowhat)
+						return
+					switch(dowhat)
+						if("tractor")
+							fire_tractor(object)
+						if("hail")
+							hail(object)
+						if("tractor-cancel")
+							release_tractor()
+			return ..() //Only tactical can fire.
+	var/list/modifiers = params2list(params)
+	if(modifiers["middle"])
+		if(istype(object, /obj/structure/overmap))
+			nav_target = object
+			return
+	if(modifiers["shift"])
+		var/atom/obj = object
+		return obj.examine(mob)
+	if(charging)
+		return
+	if(weapons_ready && mob == tactical && istype(object, /obj/structure/overmap))
+		fire(object)
+		return
+
+/obj/structure/overmap/proc/onMouseDrag(src_object, over_object, src_location, over_location, params, mob)
+	return ..()
+
+/obj/structure/overmap/proc/onMouseUp(object, location, params, mob/M)
+	return ..()
```

Comment:
why are you even defining these two if they just return ..()?

Reply:
They need to be defined or it runtimes. It's like create eye etc. on adv. cams